snowpack/src/config.ts

```diff
@@ -49,7 +49,7 @@ const DEFAULT_CONFIG: Partial<SnowpackConfig> = {
     output: 'dashboard',
     fallback: 'index.html',
     hmrDelay: 0,
-    hmrPort: 12321,
```

Comment:
If we set default hmrPort to 12321, we cannot distinguish between `snowpack dev` and `snowpack dev --hmrPort=12321`. Because we use port of devServer as hmrEngine's port while using `snowpack dev`, but we use 12321 as hmrEngine's port while using `snowpack dev --hmrPort=12321`.

Reply:
@FredKSchott  There is another way to implement it. we can add a `DEFAULT_BUILD_CONFIG` in [this file](https://github.com/pikapkg/snowpack/blob/master/snowpack/src/config.ts#L37) and merge `DEFAULT_BUILD_CONFIG` to config if snowpack is in build mode. The code of `DEFAULT_CONFIG` and `DEFAULT_BUILD_CONFIG` is as below.
```
const DEFAULT_CONFIG = {
  // other default config
  devOptions: {
     hmrPort: undefined,
  }
}

const DEFAULT_BUILD_CONFIG = {
  devOptions: {
     hmrPort: 12321,
  }
}
```
I prefer this way since we usually need to set different configuration according to whether snowpack runs in dev mode or build mode. 

I would like to use this way to modify this pr if you agree with it 😁.

Reply:
Ah right, of course. Okay, let's keep as it is now for this PR. If you'd like to create a follow up PR to simplify this config further, that would be great!